Source/AwsCommonRuntimeKit/io/TLSContextOptions.swift

```diff
@@ -9,10 +9,46 @@ public class TLSContextOptions: CStruct {
         TLSContextOptions()
     }
 
-    public static func makeMtlsPkcs12FromPath(
-        path: String,
+    /// Initializes TlsContextOptions for mutual TLS (mTLS), with client certificate and private key in the PKCS#12 format.
+    ///
+    /// NOTE: This only works on Apple devices.
```

Comment:
`@available` directive? 
Pros:
- Compile time safety instead of runtime error

Cons:
- Consumers of this API will have to do IF/ELSE at compile time, although they should probably do that anyhow if it is a cross-platform application.